libs/@guardian/libs/src/storage/README.md

```diff
@@ -119,6 +119,31 @@ Removes all items from storage.
 storage.local.clear();
 ```
 
+## `keys()`
+
+Returns: `string[] | undefined`
+
+Get all of the keys in storage.
+
+### Notes
+
+This is not a method in the native API, and gets all keys by iterating over the
+entire storage object. It is therefore potentially subject to race conditions
+and if other code is modifying storage at the same time then the results may be
+inaccurate.
+
+This method depends on the `key` method on the native API, which gets keys by
+index. But the _order_ of keys in storage is dependent on user-agent
+implementation, and so you should not rely on the order of keys returned by this
+method. (See https://developer.mozilla.org/en-US/docs/Web/API/Storage/key for
+more.)
```

Comment:
```suggestion
```

Given this describes the implementation, rather than the feature, I'd be inclined to leave this information out of the API description. 

One of the benefits of an abstracted API like this is that you can use it without worrying about what's happening below the surface. You only really need to know that stuff when working on (rather than with) the API itself.

putting this info at the point at which it's used could be very handy, but at this point is maybe distracting rather than informative?